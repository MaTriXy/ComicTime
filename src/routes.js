export const INDEX = "/";
export function publicationDetail(id) {
  return `/${id}`;
}
