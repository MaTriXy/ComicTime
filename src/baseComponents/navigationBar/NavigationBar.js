import React from "react";
import PropTypes from "prop-types";
import { AppBar, IconButton } from "material-ui";
import NavigationArrowBack from "material-ui/svg-icons/navigation/arrow-back";
import { withRouter } from "react-router";
import { connect } from "react-redux";
import { INDEX } from "../../routes";
import ComicTimeLogo from "../comicTimeLogo/ComicTimeLogo";

const style = {
  position: "fixed",
  top: 0
};

class NavigationBar extends React.Component {
  constructor(props) {
    super(props);
    this.onLeftIconButtonClick = this.onLeftIconButtonClick.bind(this);
  }

  render() {
    return (
      <div>
        <AppBar
          title={this.props.title}
          onLeftIconButtonClick={this.onLeftIconButtonClick}
          iconElementLeft={this.leftIcon()}
          style={style}
        />
      </div>
    );
  }

  leftIcon() {
    if (this.props.showBackButton) {
      return (
        <IconButton>
          <NavigationArrowBack />
        </IconButton>
      );
    } else {
      return (
        <IconButton>
          <ComicTimeLogo />
        </IconButton>
      );
    }
  }

  onLeftIconButtonClick() {
    if (this.props.showBackButton === true) {
      this.props.history.goBack();
    } else {
      this.props.history.go(INDEX);
    }
  }
}

NavigationBar.propTypes = {
  title: PropTypes.string.isRequired,
  showBackButton: PropTypes.bool,
  history: PropTypes.object
};

function mapStateToProps() {
  return {};
}

function mapDispatchToProps() {
  return {};
}

export const PureNavigationBar = NavigationBar;
export default withRouter(
  connect(mapStateToProps, mapDispatchToProps)(PureNavigationBar)
);
