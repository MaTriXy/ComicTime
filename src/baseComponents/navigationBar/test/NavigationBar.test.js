import { INDEX } from "../../../routes";

jest.mock("../../../i18n/i18n");
jest.mock("../../comicTimeLogo/ComicTimeLogo", () => "ComicTimeLogo");

import React from "react";
import { PureNavigationBar } from "../NavigationBar";
import {
  mockHistory,
  mountWithMuyThemeProvider,
  shallowComponent,
  shallowComponentAsJson
} from "../../../testUtils/enzyme";

const anyTitle = "Title";

describe("NavigationBar", () => {
  it("shows the title configured as parameter", () => {
    const component = shallowComponent(<PureNavigationBar title={anyTitle} />);
    expect(component.find("AppBar").get(0).props.title).toEqual(anyTitle);
  });

  it("renders the navigation bar using the app logo as left icon and the configured title by default", () => {
    const component = shallowComponentAsJson(
      <PureNavigationBar title={anyTitle} />
    );
    expect(component).toMatchSnapshot();
  });

  it("shows the application logo as the default left icon", () => {
    const component = mountWithMuyThemeProvider(
      <PureNavigationBar title={anyTitle} />
    );
    expect(component.find("ComicTimeLogo").length).toEqual(1);
  });

  it("renders the navigation bar using the back button instead of the logo as left icon if the param showBackButton is true", () => {
    const component = shallowComponentAsJson(
      <PureNavigationBar title={anyTitle} showBackButton={true} />
    );
    expect(component).toMatchSnapshot();
  });

  it("shows the back button as the default left icon if the component is configured to go back", () => {
    const component = mountWithMuyThemeProvider(
      <PureNavigationBar title={anyTitle} showBackButton={true} />
    );
    expect(component.find("ComicTimeLogo").length).toEqual(0);
    expect(component.find("NavigationArrowBack").length).toEqual(2);
  });

  it("navigates back to index when clicking on the app icon", () => {
    const history = mockHistory();
    const component = mountWithMuyThemeProvider(
      <PureNavigationBar title={anyTitle} history={history} />
    );
    component.find("IconButton").simulate("click");
    expect(history.go).toBeCalledWith(INDEX);
  });

  it("navigates back to index when clicking on the app icon", () => {
    const history = mockHistory();
    const component = mountWithMuyThemeProvider(
      <PureNavigationBar
        title={anyTitle}
        history={history}
        showBackButton={true}
      />
    );
    component.find("IconButton").simulate("click");
    expect(history.goBack).toBeCalled();
  });
});
