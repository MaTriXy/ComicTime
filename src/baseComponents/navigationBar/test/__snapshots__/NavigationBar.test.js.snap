// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar renders the navigation bar using the app logo as left icon and the configured title by default 1`] = `
<div>
  <AppBar
    iconElementLeft={
      <IconButton
        disableTouchRipple={false}
        disabled={false}
        iconStyle={Object {}}
        style={
          Object {
            "padding": "1",
          }
        }
        tooltipPosition="bottom-center"
        touch={false}
      >
        <ComicTimeLogo />
      </IconButton>
    }
    iconElementRight={
      <div>
        <SearchPublicationAutoComplete />
        <IconButton
          disableTouchRipple={false}
          disabled={false}
          iconStyle={Object {}}
          onClick={[Function]}
          tooltipPosition="bottom-center"
          touch={false}
        >
          <img
            alt="githubLogo"
            src="githubLogo.svg"
          />
        </IconButton>
      </div>
    }
    onLeftIconButtonClick={[Function]}
    showMenuIconButton={true}
    style={
      Object {
        "position": "fixed",
        "top": 0,
      }
    }
    title="Title"
    zDepth={1}
  />
</div>
`;

exports[`NavigationBar renders the navigation bar using the back button instead of the logo as left icon if the param showBackButton is true 1`] = `
<div>
  <AppBar
    iconElementLeft={
      <IconButton
        disableTouchRipple={false}
        disabled={false}
        iconStyle={Object {}}
        tooltipPosition="bottom-center"
        touch={false}
      >
        <NavigationArrowBack />
      </IconButton>
    }
    iconElementRight={
      <div>
        <SearchPublicationAutoComplete />
        <IconButton
          disableTouchRipple={false}
          disabled={false}
          iconStyle={Object {}}
          onClick={[Function]}
          tooltipPosition="bottom-center"
          touch={false}
        >
          <img
            alt="githubLogo"
            src="githubLogo.svg"
          />
        </IconButton>
      </div>
    }
    onLeftIconButtonClick={[Function]}
    showMenuIconButton={true}
    style={
      Object {
        "position": "fixed",
        "top": 0,
      }
    }
    title="Title"
    zDepth={1}
  />
</div>
`;
